stdlib/public/core/UnsafePointer.swift.gyb

```diff
@@ -439,22 +437,22 @@ public struct ${Self}<Pointee>
     Builtin.bindMemory(rawPtr, count._builtinWordValue, Pointee.self)
     return UnsafeMutablePointer(rawPtr)
   }
+  
+  @available(swift, deprecated: 4.1, obsoleted: 5.0.0, message: "Swift currently only supports freeing entire heap blocks, use deallocate() instead")
+  public func deallocate(capacity _: Int) { 
+    self.deallocate()
+  }
 
-  /// Deallocates memory that was allocated for `count` instances of `Pointee`.
-  ///
-  /// The memory region that is deallocated is
-  /// `capacity * MemoryLayout<Pointee>.stride` bytes in size. The memory must
-  /// not be initialized or `Pointee` must be a trivial type.
+%  end
+
+  /// Deallocates the memory block previously allocated at this pointer.
   ///
-  /// - Parameter capacity: The amount of memory to deallocate, counted in
-  ///   instances of `Pointee`.
+  /// This pointer must be a pointer to a previously allocated memory block. 
```

Comment:
Perhaps "to the start" of a previously allocated memory block, to be completely explicit.